File: src/main/java/net/igneo/icv/enchantment/weapon/MoltenEnchantment.java
package net.igneo.icv.enchantment.weapon;

import net.igneo.icv.enchantmentActions.enchantManagers.EnchantmentManager;
import net.igneo.icv.enchantmentActions.enchantManagers.weapon.MoltenManager;
import net.igneo.icv.enchantmentActions.enchantManagers.weapon.TungstenCoreManager;
import net.minecraft.world.entity.EquipmentSlot;
import net.minecraft.world.entity.player.Player;
import net.minecraft.world.item.Item;
import net.minecraft.world.item.enchantment.EnchantmentCategory;

import java.util.Set;

public class MoltenEnchantment extends WeaponEnchantment{
    public MoltenEnchantment(Rarity pRarity, EnchantmentCategory pCategory, EquipmentSlot... pApplicableSlots) {
        super(pRarity, pCategory, pApplicableSlots);
    }

    @Override
    public boolean performBonusCheck() {
        return true;
    }

    @Override
    public EnchantmentManager getManager(Player player) {
        return new MoltenManager(player);
    }

    @Override
    public boolean allowedInCreativeTab(Item book, Set<EnchantmentCategory> allowedCategories) {
        return super.allowedInCreativeTab(book, allowedCategories);
    }
}

File: src/main/java/net/igneo/icv/enchantment/weapon/MoltenEnchantment.kt
package net.igneo.icv.enchantment.weapon

import net.igneo.icv.enchantmentActions.enchantManagers.EnchantmentManager
import net.igneo.icv.enchantmentActions.enchantManagers.weapon.MoltenManager
import net.minecraft.world.entity.EquipmentSlot
import net.minecraft.world.entity.player.Player
import net.minecraft.world.item.Item
import net.minecraft.world.item.enchantment.EnchantmentCategory

class MoltenEnchantment(pRarity: Rarity?, pCategory: EnchantmentCategory?, vararg pApplicableSlots: EquipmentSlot?) :
    WeaponEnchantment(pRarity, pCategory, pApplicableSlots) {
    override fun performBonusCheck(): Boolean {
        return true
    }

    override fun getManager(player: Player?): EnchantmentManager {
        return MoltenManager(player)
    }

    override fun allowedInCreativeTab(book: Item, allowedCategories: Set<EnchantmentCategory>): Boolean {
        return super.allowedInCreativeTab(book, allowedCategories)
    }
}
